clients/src/main/java/org/apache/kafka/common/protocol/Protocol.java

```diff
@@ -384,9 +418,12 @@
                                                                         0),
                                                               new Field("responses",
                                                                       new ArrayOf(FETCH_RESPONSE_TOPIC_V0)));
+    // Fetch response V2 is the same as fetch response V1. The version number is bumped up to indicate that the Message
```

Comment:
Currently the message format change is not reflected in the fetch response protocol. The change is in the Record class when it parses the ByteBuffer. So the FetchResponse fields actually does not change. But I agree that ideally we should define all the wire protocols in Protocol. I was planning to do it in another patch because this patch is already big.

The comment here is actually not accurate. In FetchResponse V2 we may also see message format V0, because the broker will try to avoid losing zero copy by assuming the client sending FetchRequest V2 knows how to parse message format V0.